assets/js/googlesitekit/datastore/user/feature-tours.js

```diff
@@ -121,11 +132,36 @@ const baseActions = {
 			type: RECEIVE_TOURS,
 		};
 	},
+
+	receiveLastDismissedAt( timestamp ) {
+		return {
+			type: RECEIVE_LAST_DISMISSED_AT,
+			payload: {
+				timestamp,
+			},
+		};
+	},
+
+	setLastDismissedAt( timestamp ) {
+		invariant( timestamp, 'A timestamp is required.' );
+
+		return ( function* () {
+			yield {
+				type: CACHE_LAST_DISMISSED_AT,
+				payload: { timestamp },
+			};
+			yield {
+				type: RECEIVE_LAST_DISMISSED_AT,
+				payload: { timestamp },
+			};
+		}() );
+	},
```

Comment:
This is a bit awkward to read—what's wrong with:

```suggestion
	*setLastDismissedAt( timestamp ) {
		invariant( timestamp, 'A timestamp is required.' );

		yield {
			type: CACHE_LAST_DISMISSED_AT,
			payload: { timestamp },
		}
		
		yield {
			type: RECEIVE_LAST_DISMISSED_AT,
			payload: { timestamp },
		};
	},
```

Reply:
@tofumatt the `invariant` won't throw as expected in the caller if the function is a generator. Instead it will result in a rejected promise.

Reply:
Oh right, sorry, I forgot about that 👍🏻